UI/CMakeLists.txt

```diff
@@ -406,7 +406,6 @@ target_link_libraries(obs
 if (APPLE)
 	target_link_libraries(obs
 			Qt5::MacExtras)
-	set_target_properties(obs PROPERTIES LINK_FLAGS "-pagezero_size 10000 -image_base 100000000")
```

Comment:
This needs a guard, since we require it for x86 macs

Reply:
It shouldn't be required any longer, according to the "Embedding LuaJIT" documentation (and the current obs-deps repository shows LuaJIT-2.1.0-beta3 in use):
> Important: this relates to LuaJIT 2.0 only — use LuaJIT 2.1 to avoid these complications.
If you're building a 64 bit application on macOS which links directly or indirectly against LuaJIT, you need to link your main executable with these flags:
`-pagezero_size 10000 -image_base 100000000`

I was able to launch this and run specifically on an Intel Mac as well, although I don't have any specific means to to test LuaJIT -- if you have any pointers here, I'm happy to do so.

Reply:
There are example Lua scripts in the repo: https://github.com/obsproject/obs-studio/tree/cab32e7e67c2a78d10e0961f746ee7408978fb43/UI/frontend-plugins/frontend-tools/data/scripts

Reply:
Thanks, I actually stumbled upon the example scripts about the same time -- anyway, I agree, this doesn't work with Intel, at least, not on the current release of LuaJIT.  Curious if using a later build from Git as a build-time dependency will do any better... because trying to conditionalize this will be weird (i.e. I can imagine that it'd require building OBS once for each architecture, with different linker flags, in order to be able to produce a usable universal binary).

Reply:
Confirmed (by manually installing the current HEAD commit of LuaJIT) that this works when linked without `-pagezero_size`/`-image_base` -- any chance we could pull LuaJIT from Git in obs-deps instead? There's been active development on LuaJIT recently, it seems, but the 2.1.0-beta3 release is nearly 4 years old.  This would allow scripting to work (seemingly) without any changes here, or any weird build system contortions.

Reply:
> any chance we could pull LuaJIT from Git in obs-deps instead? There's been active development on LuaJIT recently, it seems, but...

Based on what's said at [http://luajit.org/download.html](http://luajit.org/download.html):

1. The git master branch will be phased out and is pinned to the v2.0 branch. Please follow the versioned branches v2.1 or v2.0 instead.
2. Releases are only made occasionally. You're strongly encouraged to follow the git branches. ... Distros that require releases should do regular snapshots of a branch. Do not attempt to cherry-pick or backport individual changes, no matter how self-standing individual changes look (because they often are not).

To me this suggests like the change would be converting LuaJIT's archive source to the following git source in `CI/flatpak/com.obsproject.Studio.json`:

```
"type": "git",
"url": "https://luajit.org/git/luajit-2.0.git",
"branch": "v2.1",
"commit": "ec6edc5c39c25e4eb3fca51b753f9995e97215da"
```

I'd say go ahead and put it into this merge request, if your testing is suggesting that the only difference is 'less code complication' with the benefit of 'newer snapshot of the code'. "Pick a point in time commit that works for you" appears to BE the LuaJIT developers' current idea of what a release is.

Reply:
Close; I think the flatpak config would impact the non-macOS dependencies; macOS (from `full-build-macos.sh`) seems to use the obs-deps project releases to grab a prebuilt copy of dependencies. But yes, you basically summarized my point exactly as "less complication with newer snapshot" -- I actually don't know what the result of trying different linker options if attempting to build as a universal binary would be; with the change as-written, though, OBS Studio will run as a universal binary, assuming all dependencies have been updated as well.

Reply:
I've updated the deps build to use the current head of the luajit 2.1 git mirror. 

https://github.com/obsproject/obs-deps/commit/0860186f32ee9b24a81c5aa738207d12ca763e9b

Once that builds I'll set the CI for the abs-studio repo to use the new deps and this should be safe to use after that.

Reply:
Updated
https://github.com/obsproject/obs-studio/commit/9b699f65d7a2e7a170db094aca151e21d32ec65c